plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -285,60 +507,341 @@ public boolean deleteNuageVspDevice(DeleteNuageVspDeviceCmd cmd) {
+                            new NetworkOfferingVO(nuageVspSharedNetworkOfferingWithSGServiceName, "Offering for NuageVsp Shared Security group enabled networks",
+                                    Networks.TrafficType.Guest, false, false, null, null, true, NetworkOffering.Availability.Optional, null, Network.GuestType.Shared, true, true, false, false, false);
+
+                    defaultNuageVspSharedSGNetworkOffering.setState(NetworkOffering.State.Enabled);
+                    defaultNuageVspSharedSGNetworkOffering = _networkOfferingDao.persistDefaultNetworkOffering(defaultNuageVspSharedSGNetworkOffering);
+
+                    Map<Network.Service, Network.Provider> defaultNuageVspSharedSGNetworkOfferingProviders = new HashMap<Network.Service, Network.Provider>();
+                    defaultNuageVspSharedSGNetworkOfferingProviders.put(Network.Service.Dhcp, Network.Provider.NuageVsp);
+                    defaultNuageVspSharedSGNetworkOfferingProviders.put(Network.Service.SecurityGroup, Network.Provider.NuageVsp);
+                    defaultNuageVspSharedSGNetworkOfferingProviders.put(Network.Service.Connectivity, Network.Provider.NuageVsp);
+
+                    for (Network.Service service : defaultNuageVspSharedSGNetworkOfferingProviders.keySet()) {
+                        NetworkOfferingServiceMapVO offService =
+                                new NetworkOfferingServiceMapVO(defaultNuageVspSharedSGNetworkOffering.getId(), service, defaultNuageVspSharedSGNetworkOfferingProviders.get(service));
+                        _networkOfferingServiceMapDao.persist(offService);
+                        s_logger.trace("Added service for the NuageVsp network offering: " + offService);
+                    }
+                }
+            }
+        });
+    }
+
+    @DB
+    private void initNuageVspVpcOffering() {
+        //configure default Nuage VSP vpc offering
+        Transaction.execute(new TransactionCallbackNoReturn() {
+            @Override
+            public void doInTransactionWithoutResult(TransactionStatus status) {
+                if (_vpcOffDao.findByUniqueName(nuageVPCOfferingName) == null) {
+                    s_logger.debug("Creating default Nuage VPC offering " + nuageVPCOfferingName);
+
+                    Map<Network.Service, Set<Network.Provider>> svcProviderMap = new HashMap<Network.Service, Set<Network.Provider>>();
+                    Set<Network.Provider> nuageProviders = new HashSet<Network.Provider>();
+                    nuageProviders.add(Network.Provider.NuageVsp);
+                    svcProviderMap.put(Network.Service.Connectivity, nuageProviders);
+                    svcProviderMap.put(Network.Service.Dhcp, nuageProviders);
+                    svcProviderMap.put(Network.Service.StaticNat, nuageProviders);
+                    svcProviderMap.put(Network.Service.SourceNat, nuageProviders);
+                    svcProviderMap.put(Network.Service.NetworkACL, nuageProviders);
```

Comment:
Could this service provider mapping be declared as a constant and initialized in a ``static`` block?